source/fundamentals/builders/aggregates.txt

```diff
@@ -11,3 +11,501 @@ Aggregates Builders
+   :dedent:
+
+Projecting Computed Fields
+++++++++++++++++++++++++++
+
+The ``$project`` stage can project computed fields as well.
+
+This example creates a pipeline stage that projects the ``rated`` field
+into a new field called ``rating``, effectively renaming the field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: begin computed
+   :end-before: end computed
+   :language: java
+   :dedent:
+
+
+Sample
+------
+
+Use the ``sample()`` method to create a :manual:`$sample </reference/operator/aggregation/sample/>`
+pipeline stage to randomly select documents from input.
+
+This example creates a pipeline stage that randomly selects 5 documents:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin sample
+   :end-before:  // end sample
+   :language: java
+   :dedent:
+
+
+
+Sort
+----
+
+Use the ``sort()`` method to create a  :manual:`$sort </reference/operator/aggregation/sort/>`
+pipeline stage to sort by the specified criteria.
+
+.. tip::
+
+   Though the sort criteria can be an instance of any class that
+   implements ``Bson``, it's convenient to combine with use of :ref:`Sorts <sorts-builders>`.
+
+This example creates a pipeline stage that sorts in descending order according
+to the value of the ``year`` field and then in ascending order according to the
+value of the ``title`` field:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin sort
+   :end-before: // end sort
+   :language: java
+   :dedent:
+
+
+Skip
+----
+
+Use the ``skip()`` method to create a :manual:`$skip </reference/operator/aggregation/skip/>`
+pipeline stage to skip over the specified number of documents that pass into the stage.
+
+This example creates a pipeline stage that skips the first ``5`` documents:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin skip
+   :end-before: // end skip
+   :language: java
+   :dedent:
+
+Limit
+-----
+
+Use the :manual:`$limit </reference/operator/aggregation/limit/>` pipeline stage
+to limit the number of documents passed to the next stage.
+
+This example creates a pipeline stage that limits the number of documents to ``10``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin limit
+   :end-before: // end limit
+   :language: java
+   :dedent:
+
+Lookup
+------
+
+Use the ``lookup()`` method to create a :manual:`$lookup </reference/operator/aggregation/lookup/>`
+pipeline stage to perform joins and uncorrelated sub-queries between two collections.
+
+Left Outer Join
++++++++++++++++
+
+This example creates a pipeline stage that performs a left outer
+join between the ``movies`` and ``comments`` collections. It joins the
+``_id`` field to the ``movie_id`` field and outputs the
+results in the ``comments`` field:
```

Comment:
Suggestion: Because multiple things are happening, I think it would be easier to follow along if the steps are bulleted.

Applies to all instances that have multiple steps.

Reply:
I may be too close to identify what the multiple steps are on the other stages. Could you point at them?